apps/projects/routes.py

```diff
@@ -1074,104 +1149,153 @@ def get(self, project_id):
-                'Content-Type': doc.get("mime_type"),
+                'Content-Type': self._project.get("mime_type"),
             }
             # get a stack of bytes push to client
-            stream = app.fs.get_range(doc['storage_id'], start, chunksize)
+            stream = app.fs.get_range(self._project['storage_id'], start, chunksize)
             res = make_response(stream)
             res.headers = headers
             return res, 206
 
         headers = {
             'Content-Length': length,
-            'Content-Type': 'video/mp4',
+            'Content-Type': self._project.get("mime_type"),
         }
-        stream = app.fs.get(doc.get('storage_id'))
+        stream = app.fs.get(self._project.get('storage_id'))
         res = make_response(stream)
         res.headers = headers
         return res, 200
 
 
-class GetRawThumbnail(MethodView):
-    SCHEME_THUMBNAIL = {
-        'index': {
-            'type': 'integer',
-            'required': False,
-            'empty': True,
-            'coerce': int,
-            'min': 0,
-        },	
-    }
+class GetRawPreviewThumbnail(MethodView):
 
     def get(self, project_id):
         """
-        Get thumbnail
+        Get preview thumbnail file
         ---
         parameters:
         - in: path
           name: project_id
           type: string
           required: True
           description: Unique project id
-        - name: index
-          in: query
-          type:
-          - integer
-          - string
-          description: index of thumbnail to get, leave empty to get preview thumbnail
         produces:
           - image/png
         responses:
           200:
-            description: OK
-            schema:
-              type: file
+            description: preview thumbnail image
+            content:
+              image/png:
+                schema:
+                  type: string
+                  format: binary
         """
-        doc = app.mongo.db.projects.find_one_or_404({'_id': format_id(project_id)})
 
-        if not request.args:
-            preview_thumbnail = doc.get('preview_thumbnail')
-            if not preview_thumbnail:
-                raise NotFound()
-            byte = app.fs.get(preview_thumbnail.get('storage_id'))
-        else:
-            schema = check_request_schema_validity(request.args.to_dict(), self.SCHEME_THUMBNAIL, allow_unknown=True)
-            index = schema['index']
-            thumbnails = next(iter(doc['thumbnails'].values()), [])
-            if len(thumbnails) < index + 1:
-                raise NotFound()
-            byte = app.fs.get(thumbnails[index]['storage_id'])
+        if not self._project['thumbnails']['preview']:
+            raise NotFound()
+
+        byte = app.fs.get(self._project['thumbnails']['preview']['storage_id'])
+        res = make_response(byte)
+        res.headers['Content-Type'] = self._project['thumbnails']['preview']['mimetype']
+        return res
```

Comment:
I think it can use some helper to send binary data for given storage item, so when there is some cdn/cache in place we can use that and to avoid duplication